### PR TITLE
improvement: add jdk indexing

### PR DESCRIPTION
**Pull request by kasiaMarek**
resolves: https://github.com/scalameta/metals/issues/6304

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/SqlSharedIndices.scala [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package scala.meta.internal.metals
+
+import java.nio.file.Paths
+
+import scala.meta.io.AbsolutePath
+
+class SqlSharedIndices
+    extends H2ConnectionProvider(
+      () => AbsolutePath(Paths.get(sys.props("user.home"))).resolve(".metals"),
+      "metals-shared",
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/SqlSharedIndices.scala:10]: Could you used names parameters? Would be useful to see what are these strings used for.

**Review by tgodzik — APPROVED** [review]
Looks good! I think this should be really useful. The only question is whether it works if user uses multiple JDKs on one machine?

**Comment by kasiaMarek** [comment]
> The only question is whether it works if user uses multiple JDKs on one machine?

Why wouldn't it work? Jars are identified by `md5` that is calculated on the path, size and last modified attribute. And we always get information out for specific jar (identified by the path).

**Comment by tgodzik** [comment]
> > The only question is whether it works if user uses multiple JDKs on one machine?
> 
> Why wouldn't it work? Jars are identified by `md5` that is calculated on the path, size and last modified attribute. And we always get information out for specific jar (identified by the path).

Just making sure :sweat_smile: We could later add the same for Scala jars

**Review by tgodzik — APPROVED** [review]
LGTM Thanks!

**Pull request merged**